src/main/java/org/broadinstitute/hellbender/tools/exome/cnlohcaller/CNLOHCaller.java

```diff
@@ -0,0 +1,752 @@
+            for (int k = 0; k < data[0][0].length; k++) {
+                result[j][k] = 0;
+                for (int i = 0; i < data.length; i++) {
+                    result[j][k] += data[i][j][k];
+                }
+            }
+        }
+
+        return result;
+    }
+
+    private double[] sumOverSecondDimension(final double[][][] data) {
+        final double [] result = new double[data[0].length];
+        for (int j = 0; j < data[0].length; j++) {
+            result[j] = 0;
+            for (int i = 0; i < data.length; i++) {
+                for (int k = 0; k < data[0][0].length; k++) {
+                    result[j] += data[i][j][k];
+                }
+            }
+        }
+
+        return result;
+    }
+
+    private double[] sumOverThirdDimension(final double[][][] data) {
+        final double [] result = new double[data[0][0].length];
+        for (int k = 0; k < data[0][0].length; k++) {
+            result[k] = 0;
+            for (int i = 0; i < data.length; i++) {
+                for (int j = 0; j < data[0].length; j++) {
+                    result[k] += data[i][j][k];
+                }
+            }
+        }
+
+        return result;
+    }
+
+    // Assumes that the pis parameter is a SortedMap.
+    protected double[][][] calcE_zsk_vsm_wsn(final double[] E_ln_phi_k, final SortedMap<Integer, Double> pis,
+                                             final double[] rhos, final double mafCredibleMode,
+                                             final double mafCredibleLow, final double mafCredibleHigh,
+                                             final double crCredibleMode, final double crCredibleLow,
+                                             final double crCredibleHigh, final double lambda) {
+
+
+        final int[] mVals = pis.keySet().stream().mapToInt(Integer :: intValue).toArray();
+        final int[] nVals = pis.keySet().stream().mapToInt(Integer :: intValue).toArray();
+
+        final double[][][] result = new double[rhos.length][mVals.length][nVals.length];
+        double total = 0.0;
+        for (int rhoi = 0; rhoi < rhos.length; rhoi++) {
+            for (int mi = 0; mi < mVals.length; mi++) {
+                for (int ni = 0; ni < nVals.length; ni++) {
+
+                    final double rho = rhos[rhoi];
+                    final int m = mVals[mi];
+                    final int n = nVals[ni];
+
+                    if ((rho < 0) || (rho > 1)) {
+                        result[rhoi][mi][ni] = 0;
+                    } else if ((rho == 0) && ((m != 1) || (n != 1))) {
+                        result[rhoi][mi][ni] = 0;
+                    } else if ((rho > 0) && (rho < rhoThreshold)) {
+                        result[rhoi][mi][ni] = 0;
+                    } else if ((rho != 0) && ((m == 1) && (n == 1))) {
+                        result[rhoi][mi][ni] = 0;
+                    } else {
+                        result[rhoi][mi][ni] = Math.exp(E_ln_phi_k[rhoi]) *
```

Comment:
Instead of calculating and storing `E_ln_phi_k` and exponentiating, why not calculate and store `Math.exp(E_ln_phi_k)`?  This would let you name the variable (and associated methods) `effectivePhi`, which is more descriptive and encapsulating.